spec.bs

```diff
@@ -0,0 +1,473 @@
+    1. If |initiatorWindow| is not an [=environment settings object=] whose [=environment settings object/global object=] is a {{Window}} object, then return.
+    1. If |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> is not a [=secure context=], then return.
+    1. For each feature |f| in « "browsing-topic", "interest-cohort" »:
+        1. Run the <a href="https://www.w3.org/TR/permissions-policy-1/#algo-should-request-be-allowed-to-use-feature">Should request be allowed to use feature?</a> algorithm with <var ignore=''>feature</var> set to |f| and <var ignore=''>request</var> set to |request|. If the algorithm returns false, then return.
+
+        Note: the above algorithm should include the <a href="https://github.com/w3c/webappsec-permissions-policy/pull/499">pending update</a>, i.e. the |request| should be considered to contain the equivalent opt-in flags for both "browsing-topic" and the "interest-cohort" feature.
+    1. If the user preference setting disallows the access to topics from |topicsCallerContext|'s [=topics caller context/caller origin=] or |topicsCallerContext|'s [=topics caller context/top level context domain=], then return.
+    1. Let |topicsCallerContext| be a [=topics caller context=] with default initial field values.
+    1. Set |topicsCallerContext|'s [=topics caller context/caller origin=] to |requestOrigin|.
+    1. Set |topicsCallerContext|'s [=topics caller context/top level context domain=] to |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a>'s [=top-level browsing context=]'s [=active document=]'s {{Document/domain}}.
+    1. Let |topics| be the result from running the [=topics for caller=] algorithm, with |topicsCallerContext| as input.
+    1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-lists">List</a>.
+    1. For each |topic| in |topics|:
+        1. Let |topicItem| be an Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-integers">Integer</a> with value |topic|'s {{BrowsingTopic/topic}}.
+        1. Let |topicParameters| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parameters">Parameters</a>.
+        1. Set |topicParameters|["version"] to |topic|'s {{BrowsingTopic/version}}.
+        1. Set |topicParameters|["config_version"] to |topic|'s {{BrowsingTopic/configVersion}}.
+        1. Set |topicParameters|["model_version"] to |topic|'s {{BrowsingTopic/modelVersion}}.
+        1. Set |topicParameters|["taxonomy_version"] to |topic|'s {{BrowsingTopic/taxonomyVersion}}.
+        1. Associate |topicParameters| with |topicItem|.
+        1. Insert |topicItem| to |headerStructuredFields|.
+    1. <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-serializing-structured-fiel">Serialize</a> |headerStructuredFields| into [=string=] |headerValue|.
```

Comment:
Use https://fetch.spec.whatwg.org/#concept-header-list-set-structured-header instead of the next two steps.